src/backend/parts/Messages.js

```diff
@@ -170,14 +159,98 @@ const MessageListItem = ( { post, onClick, isSelected, isFirst } ) => (
 	</Button>
 );
 
-const MessageList = ( {
-	posts,
-	selectedPost,
-	onPostSelect,
-	loadMoreButton,
+const Pagination = ( {
+	currentPage,
+	onPageChange,
+	postsPerPage,
+	totalPosts,
+	setUpsellOpen,
 } ) => {
+	const totalPages = Math.ceil( totalPosts / postsPerPage );
+	const hasNext = currentPage < totalPages;
+	const hasPrev = currentPage > 1;
+	const isFreePlan = ! window.hyve.hasPro;
+
+	const handlePageChange = ( page ) => {
+		if ( isFreePlan ) {
+			setUpsellOpen( true );
+			return;
+		}
+
+		if ( page >= 1 && page <= totalPages ) {
+			onPageChange( page );
+		}
+	};
+
 	return (
-		<div className="col-span-6 xl:col-span-2 border-r-gray-300 border-r-[0.5px] border-solid max-h-[672px] overflow-scroll">
+		<div className="flex items-center justify-between py-2 px-4">
+			<div className="text-[#50575e] text-[14px] font-medium flex-1 text-end px-5">
+				{ totalPosts } { __( 'items', 'hyve-lite' ) }
+			</div>
+			<div className="flex items-center">
+				<button
+					onClick={ () => handlePageChange( 1 ) }
+					disabled={ ! hasPrev && ! isFreePlan }
+					className={ `flex items-center justify-center w-8 h-8 border border-[#c3c4c7] ${
+						! hasPrev && ! isFreePlan
+							? 'bg-[#f0f0f1] text-[#a7aaad] cursor-default'
+							: 'bg-white text-[#2271b1] hover:bg-[#f6f7f7] hover:border-[#8c8f94]'
+					} rounded-sm focus:outline-none mr-1` }
+					aria-label={ __( 'Go to the first page', 'hyve-lite' ) }
+				>
+					<span className="text-lg">«</span>
+				</button>
```

Comment:
You can combine this into a class named `PaginationButton`. It will be used like:

```jsx
<PaginationButton
        onClick={}
        disabled={}
        aria-label={}
>
        <span className="text-lg">«</span>
</PaginationButton>
```

And the class style will be controlled by `disabled`